src/services/users.service.ts

```diff
@@ -52,6 +52,17 @@ class UserService {
     const deleteUserData = await this.users.delete({ where: { id: userId }, select: { email: true, id: true } });
     return deleteUserData;
   }
+
+  public async manageGroups(userId: number, data: any) {
+    if (isEmpty(data)) throw new HttpException(400, "Groups don't exist");
+    
+    const groups = await this.users.update({
+      where: { id: userId },
+      data
+    })
+
+    return groups;
+  }
 }
```

Comment:
1. Array Data Type: It seems like data is intended to be used to update the user records, possibly an array of new groups to assign to the user. However, the data type for data isn't specified which could lead to mistakenly passing the wrong data type. It would be recommended to define an interface for the data that you're passing.

2. Check object properties: The isEmpty function will pass if data is not null, undefined, or an empty object ({}). However, this doesn't necessarily mean that data contains the needed fields. Validate the fields inside data. If you're using class-validator, this will be simple since it validates objects based on predefined classes.

3. Handler Error: Instead of throwing HttpException specifically, it would be better to throw generic errors from your services, and then handle them in your controller classes where you can throw HttpException. This approach is better for the separation of concerns.

4. API Consistency: If you're using Prisma, it's a good idea to stick to Prisma's API rather than using mixture of native JavaScript and Prisma. Prisma should be able to handle updates and return the updated record using the include property

5. Function Return Type: The function manageGroups doesn't have a definition for the return type. This should be defined for better code understanding and robustness.

6. Error Messages: Using generic error messages might be hard to understand or solve. Providing more context might help

7. Variable Naming: The groups variable that stores the return value of this.users.update() doesn't entirely match the data it holds, this might cause confusion. Hence, you might want to change this to something descriptive related to user or userData.

8. Meaningful condition checks: Here, isEmpty(data) just checks if the data is there or not. You may want to add further checks for validation of the data to ensure correct usage.

Here is example of new version:
```
interface UpdateUserData {
  groups?: Prisma.GroupUpdateManyMutationInput;
  // other fields...
}

public async manageUserGroups(userId: number, updateData: UpdateUserData): Promise<User|null> {
  if (isEmpty(updateData)) throw new Error("No update data provided");

  const updatedUser = await this.users.update({
    where: { id: userId },
    data: updateData,
    include: {
      groups: true,
    },
  });

  return updatedUser;
}
```